langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java

```diff
@@ -48,19 +51,55 @@
 public class MistralAiChatModel implements ChatModel {
 
     private final MistralAiClient client;
-    private final String modelName;
-    private final Double temperature;
-    private final Double topP;
-    private final Integer maxTokens;
     private final Boolean safePrompt;
     private final Integer randomSeed;
-    private final ResponseFormat responseFormat;
 
     private final Integer maxRetries;
     private final Set<Capability> supportedCapabilities;
 
+    private ChatRequestParameters defaultRequestParameters;
+
+    private MistralAiChatModel(MistralAiChatModelBuilder builder) {
+        this.client = MistralAiClient.builder()
+                .httpClientBuilder(builder.httpClientBuilder)
+                .baseUrl(getOrDefault(builder.baseUrl, "https://api.mistral.ai/v1"))
+                .apiKey(builder.apiKey)
+                .timeout(getOrDefault(builder.timeout, Duration.ofSeconds(60)))
+                .logRequests(getOrDefault(builder.logRequests, false))
+                .logResponses(getOrDefault(builder.logResponses, false))
+                .build();
+
+        this.safePrompt = builder.safePrompt;
+        this.randomSeed = builder.randomSeed;
+        this.maxRetries = getOrDefault(builder.maxRetries, 2);
+        this.supportedCapabilities = getOrDefault(builder.supportedCapabilities, Set.of());
+        this.defaultRequestParameters = initDefaultRequestParameters(builder);
+    }
+
+    private ChatRequestParameters initDefaultRequestParameters(MistralAiChatModelBuilder builder) {
+        ChatRequestParameters commonParameters;
+        if (builder.defaultRequestParameters != null) {
+            validate(builder.defaultRequestParameters);
+            commonParameters = builder.defaultRequestParameters;
+        } else {
+            commonParameters = DefaultChatRequestParameters.EMPTY;
+        }
+
+        return DefaultChatRequestParameters.builder()
+                .modelName(ensureNotBlank(getOrDefault(builder.modelName, commonParameters.modelName()), "modelName"))
```

Comment:
```suggestion
                .modelName(getOrDefault(builder.modelName, commonParameters.modelName()))
```
model name can also be provided when calling `chat(ChatRequest)`, so no need to fail so early